src/solvers/smt2/smt2_conv.cpp

```diff
@@ -3326,15 +3326,16 @@ void smt2_convt::convert_plus(const plus_exprt &expr)
         "one of the operands should have pointer type");
 
       mp_integer element_size;
-      if(expr.type().subtype().id() == ID_empty)
+      if(to_pointer_type(expr.type()).base_type().id() == ID_empty)
       {
         // This is a gcc extension.
         // https://gcc.gnu.org/onlinedocs/gcc-4.8.0/gcc/Pointer-Arith.html
         element_size = 1;
       }
       else
       {
-        auto element_size_opt = pointer_offset_size(expr.type().subtype(), ns);
+        auto element_size_opt =
+          pointer_offset_size(to_pointer_type(expr.type()).base_type(), ns);
```

Comment:
A temporary may be useful here to avoid the repeated `to_pointer_type`.

Reply:
done